### PR TITLE
feat: injection on output types (outjection)

### PR DESCRIPTION
**Pull request by Natoandro**
- [x] Solve [MET-140](https://linear.app/metatypedev/issue/MET-140/gate-computestage-processing-from-context-should-evaluate-in-all-cases)
  - Enable injection on output
- [x] Solve [MET-47](https://linear.app/metatypedev/issue/MET-47/gate-from-parent-for-eitherunion)
  - test from parent injection for either/union types

#### Migration notes
- _N/A_

---

- [x] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by michael-0acf4 — DISMISSED** [review]
Nice!

**Review by luckasRanarison — DISMISSED** [review]

**Review by michael-0acf4 — DISMISSED** [review]

File: tests/injection/outjection.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        Policy.public(),
+        randomUser=deno.identity(t.struct()).extend(
+            {
+                "id": t.uuid().from_random(),
+                "age": t.integer().set(19),
+                "email": t.email().from_context("user_email"),
+                "password": t.string().from_secret(
+                    "user_password"
+                ),  ## should we allow this?
+                "createdAt": t.datetime().inject("now"),
```
Comment (Yohe-Am) [inline tests/injection/outjection.py:23]: Does metagen handle this well btw? I.e. do we need to generate separate types without the injeted fields for the FDK/custom function implementers? I don't think it does, should I add it to the meagen papercuts tickeT?

File: tests/injection/outjection.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    random = RandomRuntime(seed=1)
+
+    g.expose(
+        Policy.public(),
+        randomUser=deno.identity(t.struct()).extend(
+            {
+                "id": t.uuid().from_random(),
+                "age": t.integer().set(19),
+                "email": t.email().from_context("user_email"),
+                "password": t.string().from_secret(
```
Comment (Yohe-Am) [inline tests/injection/outjection.py:20]: tangent: do we have support or will we want to add support for optional injetions? I.e. only inject if no value is found on output?

File: tests/injection/outjection.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+@typegraph()
+def outjection(g: Graph):
+    deno = DenoRuntime()
+    random = RandomRuntime(seed=1)
+
+    g.expose(
+        Policy.public(),
+        randomUser=deno.identity(t.struct()).extend(
```
Comment (Yohe-Am) [inline tests/injection/outjection.py:15]: tangent: would it be sensible to move `extend` onto `t.Struct`?

**Review by Yohe-Am — APPROVED** [review]

File: tests/injection/outjection.py (hunk 3 above)
Comment (Natoandro) [reply tests/injection/outjection.py:15]: `extend` was originally on `t.struct`, but it is currently missing on the TS SDK. On `t.func`, it applies on the output type (if it is a `t.struct`). Main reason was the heavy use of `deno.identity` in the tests, which btw won't play well with injections after this PR...

File: tests/injection/outjection.py (hunk 2 above)
Comment (Natoandro) [reply tests/injection/outjection.py:20]: Interesting... We can explore that...

File: tests/injection/outjection.py (hunk 1 above)
Comment (Natoandro) [reply tests/injection/outjection.py:23]: For injection on output types ("outjections"), it think it is not needed. But, it is indeed necessary for injections..

File: tests/injection/outjection.py (hunk 3 above)
Comment (Yohe-Am) [reply tests/injection/outjection.py:15]: https://linear.app/metatypedev/issue/MET-778/move-extend-to-tstructand-documentation

File: tests/injection/outjection.py (hunk 1 above)
Comment (Yohe-Am) [reply tests/injection/outjection.py:23]: But wouldn't the type system complain if you don't provide these outjected values in your custom function responses?

File: tests/injection/outjection.py (hunk 2 above)
Comment (Yohe-Am) [reply tests/injection/outjection.py:20]: https://linear.app/metatypedev/issue/MET-780/optional-injections

File: tests/injection/outjection.py (hunk 1 above)
Comment (Natoandro) [reply tests/injection/outjection.py:23]: Ah, you are right... It is needed for the FDK...

**Review by luckasRanarison — APPROVED** [review]

**Pull request merged**